clients/da-vinci-client/src/main/java/com/linkedin/davinci/ingestion/isolated/IsolatedIngestionServer.java

```diff
@@ -460,14 +449,16 @@ public void setPartitionToBeSubscribed(String topicName, int partition) {
 
   // Check if topic partition is being subscribed.
   public boolean isPartitionSubscribed(String topicName, int partition) {
-    if (!topicPartitionSubscriptionMap.containsKey(topicName)) {
-      return false;
-    }
-    Map<Integer, AtomicBoolean> partitionSubscriptionMap = topicPartitionSubscriptionMap.get(topicName);
-    if (!partitionSubscriptionMap.containsKey(partition)) {
+    AtomicBoolean subscription =
+        getTopicPartitionSubscriptionMap().getOrDefault(topicName, Collections.emptyMap()).get(partition);
+    if (subscription == null) {
       return false;
     }
-    return partitionSubscriptionMap.get(partition).get();
+    return subscription.get();
```

Comment:
Thanks for fixing the race here. `cleanupTopicPartitionState` seems to have a similar issue. Feel free to fix it together or in another one, just let you know